session.go

```diff
@@ -131,9 +129,22 @@ func startSession(conn net.Conn, windowSize int, maxPadding int, ackOnFirst bool
 	atomic.AddInt64(&openSessions, 1)
 	ops.Go(s.sendLoop)
 	ops.Go(s.recvLoop)
+	if clientInitMsg != nil {
+		s.sendClientInitMsg(clientInitMsg)
+	}
 	return s, nil
 }
 
+func (s *session) sendClientInitMsg(clientInitMsg []byte) {
+	// Client init message is already encrypted
+	copy(s.sendSessionFrame, clientInitMsg)
```

Comment:
Using `s.sendSessionFrame` here creates a race condition because sendSessionFrame is used inside of the send loop.

Reply:
Moving the clientInitMsg back into the send loop would fix the race condition.

Reply:
Fixed at https://github.com/getlantern/lampshade/pull/34/commits/aa8a7f7a7baba28a547c723c29b64d01c32d445b by calling `sendClientInitMsg` before starting the loops.